core/src/main/java/org/apache/calcite/rel/rules/SingleValuesOptimizationRules.java

```diff
@@ -245,24 +256,29 @@ protected void onMatch(RelOptRuleCall call, Values values,
         rexNodes = new ArrayList<>(values.tuples.get(0));
       }
       RelBuilder relBuilder = call.builder();
+
+      BiFunction<RexNode, List<RexNode>, List<RexNode>> transformer =
+          getRexTransformer(relBuilder.getRexBuilder(), join.getJoinType());
+
       RelNode transformed =
           new SingleValuesRelTransformer(join, rexNodes, other,
-              predicate, isLeft, getRexTransformer(relBuilder.getRexBuilder(),
-              join.getJoinType())).transform(relBuilder);
+              transformableCheck, isOnLefSide, transformer).transform(relBuilder);
```

Comment:
At least with IntelliJ I have trouble stepping in "transform" here.
So I would even allocate a temp variable for the transformer and invoke `transform` in a separate statement.